### PR TITLE
Fixed "AttributeError: list object has no attribute strip" with quorum queues and failover brokers

### PR DESCRIPTION
**Pull request by Nusnus**

### CODE REVIEW

File: t/unit/worker/test_native_delayed_delivery.py [hunk 1]
```diff
@@ -74,26 +74,36 @@ def test_start_native_delayed_delivery_fanout_exchange(self, caplog):
...
+            ([], ValueError, "broker_url configuration is empty"),
+            (123, ValueError, "broker_url must be a string or list"),
+            (["amqp://", 123, None, "amqp://"], ValueError, "All broker URLs must be strings"),
+        ],
+    )
+    def test_validate_broker_urls_invalid(self, broker_urls, exception_type, exception_match):
         delayed_delivery = DelayedDelivery(Mock())
-
-        with pytest.raises(ValueError, match="No valid broker URLs found in configuration"):
-            delayed_delivery._validate_broker_urls("  ;  ;  ")
```
Comment (jayaddison) [inline t/unit/worker/test_native_delayed_delivery.py:90]: I think this test case for multiple empty broker URLs would begin failing with the hotfixed code.  Could we retain it in the parametrized tests?

Comment (jayaddison) [reply t/unit/worker/test_native_delayed_delivery.py:90]: I've prepared a branch with a possible fixup - please let me know whether a PR for this would be welcome.

Comment (Nusnus) [reply t/unit/worker/test_native_delayed_delivery.py:90]: @jayaddison 
> I think this test case for multiple empty broker URLs would begin failing with the hotfixed code. Could we retain it in the parametrized tests?

Not sure I understand - the CI passes.

> I've prepared a branch with a possible fixup - please let me know whether a PR for this would be welcome.

Thank you! - let’s make sure we first understand what’s exactly is that we’re trying to fix :)

Comment (Nusnus) [reply t/unit/worker/test_native_delayed_delivery.py:90]: @jayaddison 
> I think this test case for multiple empty broker URLs would begin failing with the hotfixed code. Could we retain it in the parametrized tests?

Technically you’re right. The `"  ;  ;  “` test case _will_ fail.
That being said, an empty `broker_url` will default to `amqp` automatically so these cases _should never happen_. I did add some checks because we know that things that shouldn’t happen do happen, but this is a very rare edge case (`"  ;  ;  “`).

These tests ensure that the private helper method covers the basic input cases, but the delayed delivery feature is not responsible for the broker URL’s validity.

I’ve removed the use of `strip` to make the broker validation just a bit more flexible to balance basic checks with not taking full responsibility over `broker_url`.

> I've prepared a branch with a possible fixup - please let me know whether a PR for this would be welcome.

Do you want to share briefly what is your suggested change to make sure we’re aligned before opening a PR?
I don’t mind improving the coverage as long as we keep it simple.

Thank you!

Comment (jayaddison) [reply t/unit/worker/test_native_delayed_delivery.py:90]: Thanks @Nusnus!

> > I think this test case for multiple empty broker URLs would begin failing with the hotfixed code. Could we retain it in the parametrized tests?
> 
> Technically you’re right. The `" ; ; “` test case _will_ fail. That being said, an empty `broker_url` will default to `amqp` automatically so these cases _should never happen_. I did add some checks because we know that things that shouldn’t happen do happen, but this is a very rare edge case (`" ; ; “`).

I mostly agree - my line of thinking is that if there was a form of misconfiguration we could previously detect early, then let's continue to do that.
 
> These tests ensure that the private helper method covers the basic input cases, but the delayed delivery feature is not responsible for the broker URL’s validity.

:+1: 

> I’ve removed the use of `strip` to make the broker validation just a bit more flexible to balance basic checks with not taking full responsibility over `broker_url`.
> 
> > I've prepared a branch with a possible fixup - please let me know whether a PR for this would be welcome.
> 
> Do you want to share briefly what is your suggested change to make sure we’re aligned before opening a PR? I don’t mind improving the coverage as long as we keep it simple.

The two modifications I'd suggest - both small are:

  * Test coverage: restoring the `"  ;  ;  "` case into the parameterized tests (confirming the problem).
  * Code fixup: restoring the `if url.strip()` conditions into the list comprehension (restoring the previous safety mechanism).

Comment (Nusnus) [reply t/unit/worker/test_native_delayed_delivery.py:90]: @jayaddison 
> The two modifications I'd suggest - both small are:
> 
> * Test coverage: restoring the `"  ;  ;  "` case into the parameterized tests (confirming the problem).
> * Code fixup: restoring the `if url.strip()` conditions into the list comprehension (restoring the previous safety mechanism).

I think it’ll be better to have a proper global verification method, maybe in Kombu even, and then make sure it is fully covered for every edge case including the rare and exotics and use it here instead.

For now we needed a quick hotfix for the linked bug so I left the `strip`ing issue for another time.

Comment (jayaddison) [reply t/unit/worker/test_native_delayed_delivery.py:90]: Ok, thank you.
